lib/services.js

```diff
@@ -209,17 +213,31 @@ export default {
         // confirm date format
         omVal['last-modified'] = new Date().toJSON();
         omVal['content-md5'] = contentMD5;
-        // Need to complete values
-        omVal['x-amz-server-side-encryption'] = '';
+
         omVal['x-amz-server-version-id'] = '';
         // TODO: Handle this as a utility function for all object puts
         // similar to normalizing request but after checkAuth so
         // string to sign is not impacted.  This is GH Issue#89.
         omVal['x-amz-storage-class'] = 'STANDARD';
         omVal['x-amz-website-redirect-location'] = '';
-        omVal['x-amz-server-side-encryption-aws-kms-key-id'] = '';
-        omVal['x-amz-server-side-encryption-customer-algorithm'] = '';
         omVal.location = dataGetInfo;
+
+        if (cipherBundle) {
+            omVal['x-amz-server-side-encryption'] = cipherBundle.algorithm;
+            if (cipherBundle.masterKeyId) {
+                omVal['x-amz-server-side-encryption-aws-kms-key-id'] =
```

Comment:
To ensure the object signature remains consistent, let's set it to null if it's a falsey value
```javascript
omVal['x-amz-server-side-encryption-aws-kms-key-id'] = cipherBundle.masterKeyId || null
```